src/c2pa/c2pa.py

```diff
@@ -1770,66 +1768,72 @@ def sign(
                 if error:
                     raise C2paError(error)
 
-            manifest_bytes = None
             if manifest_bytes_ptr:
-                # Convert the manifest bytes to a Python bytes object
-                size = result
-                manifest_bytes = bytes(manifest_bytes_ptr[:size])
+                # Free the manifest bytes pointer if it was allocated
                 _lib.c2pa_manifest_bytes_free(manifest_bytes_ptr)
 
-            return manifest_bytes
+            return result
         finally:
             # Ensure both streams are cleaned up
             source_stream.close()
             dest_stream.close()
 
+    def sign(
+            self,
+            signer: Signer,
+            format: str,
+            source: Any,
+            dest: Any = None) -> None:
+        """Sign the builder's content and write to a destination stream.
+
+        Args:
+            format: The MIME type or extension of the content
+            source: The source stream (any Python stream-like object)
+            dest: The destination stream (any Python stream-like object)
+            signer: The signer to use
+
+        Raises:
+            C2paError: If there was an error during signing
+        """
+        # Convert Python streams to Stream objects
+        source_stream = Stream(source)
+        dest_stream = Stream(dest)
+
+        # Use the internal stream-base signing logic
+        self._sign_internal(signer, format, source_stream, dest_stream)
+
     def sign_file(self,
                   source_path: Union[str,
                                      Path],
                   dest_path: Union[str,
                                    Path],
-                  signer: Signer) -> tuple[int,
-                                           Optional[bytes]]:
+                  signer: Signer) -> int:
         """Sign a file and write the signed data to an output file.
 
         Args:
             source_path: Path to the source file
             dest_path: Path to write the signed file to
+            signer: The signer to use
 
         Returns:
-            A tuple of (size of C2PA data, optional manifest bytes)
+            Size of C2PA data
```

Comment:
Sign_file needs to return manifest_bytes so that they can be used for sidecar or uploading remote manifests.

Reply:
What is the purpose of returning the size of the manifest_bytes here?